src/anndata/experimental/__init__.py

```diff
@@ -1,21 +1,29 @@
 from __future__ import annotations
 
+import warnings
+from typing import TYPE_CHECKING
+
 from anndata._core.sparse_dataset import CSCDataset, CSRDataset, sparse_dataset
 from anndata._io.specs import IOSpec, read_elem, read_elem_as_dask, write_elem
 
-from .._types import InMemoryElem as _InMemoryElem
 from .._types import Read, ReadCallback, StorageType, Write, WriteCallback
-from .._types import RWAble as _RWAble
 from ._dispatch_io import read_dispatched, write_dispatched
 from .merge import concat_on_disk
 from .multi_files import AnnCollection
 from .pytorch import AnnLoader
 
-# Sphinx can’t find data docstrings when objects are re-exported
-InMemoryElem = _InMemoryElem
-"""An in-memory element that can be read and written, including an :class:`anndata.AnnData` objects."""
-RWAble = _RWAble
-"""A serializable object, excluding :class:`anndata.AnnData` objects i.e., something that can be stored in `uns` or `obsm`."""
+if TYPE_CHECKING:
+    from typing import Any
+
+deprecated = ["CSRDataset", "CSCDataset", "read_elem", "write_elem"]
+
+
+def __getattr__(key: str) -> Any:
+    if key in deprecated:
+        msg = f"Importing {key} from `anndata.experimental` is deprecated. Import from `anndata` directly."
+        warnings.warn(msg, FutureWarning)
+    return globals()[key]
```

Comment:
`__getattr__` is only called if the name is *not* in `globals()`, so we need to remove the import and change this: 

```suggestion
        import anndata

        return getattr(anndata, key)
    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
```

Reply:
That makes more sense! "in a module that would act like a normal __getattr__ method"